File: api-aggregator/src/main/java/com/github/alexeysol/geekregime/apiaggregator/shared/utils/sources/ApiUsersSource.java
package com.github.alexeysol.geekregime.apiaggregator.shared.utils.sources;

import com.github.alexeysol.geekregime.apicommons.models.sources.ApiPath;
import com.github.alexeysol.geekregime.apicommons.models.sources.BaseUrl;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class ApiUsersSource implements ApiPath, BaseUrl {
    @Value("${api-users.base-url}")
    private String baseUrl;

    @Value("${api-users.prefix}")
    private String apiPrefix;

    @Value("${api-users.resource}")
    private String resource;

    public String getBaseUrl() {
        return baseUrl;
    }

    public String getApiPath(int version) {
        return String.format("/%s/v%d/%s", apiPrefix, version, resource);
    }
}

File: api-users/src/main/kotlin/com/github/alexeysol/geekregime/apiusers/utils/sources/ApiUsersSource.kt
package com.github.alexeysol.geekregime.apiusers.utils.sources

import com.github.alexeysol.geekregime.apicommons.models.sources.ActiveProfile
import com.github.alexeysol.geekregime.apicommons.models.sources.ApiPath
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
data class ApiUsersSource (
    @Value("\${spring.profiles.active}")
    private val activeProfile: String,

    @Value("\${api-users.prefix}")
    private val apiPrefix: String,

    @Value("\${api-users.resource}")
    val resource: String? = null,
) : ApiPath, ActiveProfile {
    override fun getApiPath(version: Int): String = formatApiPath(apiPrefix, version, resource)

    override fun getActiveProfile(): String = activeProfile
}
